### PR TITLE
feat(TreeView): onExpandedChange returns an array of expanded IDs

### PR DESCRIPTION
**Pull request by moathabuhamad-cengage**
Issue: #1499

## What I did

- Implemented functionality for onExpandedChange to return an array of expanded items similar to how onSelectedItemChange returns selected items. 

## Checklist 
- [x] changeset has been added
- [x] Pull request description is descriptive
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [-] I have added tests that prove my fix is effective or that my feature works

## How to test
test the implementation example in 
- storybook: /treeview--complex
- documentation: api/tree-view/#change_events

### CODE REVIEW

File: packages/react-magma-dom/src/components/TreeView/useTreeView.ts [hunk 1]
```diff
@@ -360,6 +364,33 @@ export function useTreeView(props: UseTreeViewProps) {
...
+        updatedExpandedSet.delete(itemId);
+      } else {
+        updatedExpandedSet.add(itemId);
+      }
+
+      const expandedItemsArray = Array.from(updatedExpandedSet);
+
+      if (onExpandedChange) {
+        onExpandedChange(event, expandedItemsArray);
+      }
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/TreeView/useTreeView.ts:388]:
```suggestion
    onExpandedChange &&
        typeof onExpandedChange === 'function' &&
        onExpandedChange(event, expandedItemsArray);
```

File: packages/react-magma-dom/src/components/TreeView/useTreeView.ts [hunk 2]
```diff
@@ -63,8 +63,12 @@ export interface UseTreeViewProps {
   testId?: string;
   /**
    * Action that fires when an item is expanded or collapsed
+   * Return an Array list of itemIds of items that are expanded
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/TreeView/useTreeView.ts:66]: Just to match the description for onSelectedItemChange,
```suggestion
   * Return an array of itemIds of items that are expanded
   * Example: ['item0', 'item1', 'item3']
```

File: packages/react-magma-dom/src/components/TreeView/TreeView.stories.tsx [hunk 3]
```diff
@@ -209,13 +209,22 @@ export const Complex = (args: Partial<TreeViewProps>) => {
   );
   const total = selectedItems?.length || 0;
 
+  const handleExpandedChange = (event :React.SyntheticEvent) => {
+    console.log(event);
+  }
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/TreeView/TreeView.stories.tsx:214]:
```suggestion
  const handleExpandedChange = (event: React.SyntheticEvent, items) => {
    console.log(event, items);
    setExpandedItems(items);
  };
```

Then under the button group we can add something like this to keep track of what items are being passed back
```
      <Spacer axis={SpacerAxis.vertical} size={24} />
      <p>Expanded: {expandedItems?.join(', ')}</p>
```

When I did this locally, I noticed that when `initialExpandedItems` is passed, the array will always return them even after they have been collapsed:

image

Comment (moathabuhamad-cengage) [reply packages/react-magma-dom/src/components/TreeView/TreeView.stories.tsx]: done

File: packages/react-magma-dom/src/components/TreeView/TreeView.stories.tsx [hunk 4]
```diff
@@ -201,13 +203,19 @@ export const Complex = (args: Partial<TreeViewProps>) => {
...
 
   const { selected, indeterminate } = createControlledTags(
     selectedItems,
     apiRef?.current
   );
-  const total = selectedItems?.length || 0;
+  const total = selectedItems?.length ?? 0;
+
+  const handleExpandedChange = (event: React.SyntheticEvent, expandedItems: string[]) => {
+    console.log(event);
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/TreeView/TreeView.stories.tsx:216]: I think we can do without this console log

**Review by silvalaura — APPROVED** [review]

**Pull request merged**